java/integTest/src/test/java/glide/ConnectionTests.java

```diff
@@ -26,4 +39,173 @@ public void cluster_client() {
+        }
+
+        ClusterValue<String> infoResult =
+                azTestClient
+                        .info(new InfoOptions.Section[] {InfoOptions.Section.COMMANDSTATS}, ALL_NODES)
+                        .get();
+        Map<String, String> infoData = infoResult.getMultiValue();
+
+        // Check that all replicas have the same number of GET calls
+        long matchingEntries =
+                infoData.values().stream().filter(value -> value.contains(getCmdstat)).count();
+        assertEquals(nReplicas, matchingEntries);
+        azTestClient.close();
+    }
+
+    /**
+     * Test that the client with az affinity strategy will only route to the 1 replica with the same
+     * az.
+     */
+    @SneakyThrows
+    @Test
+    public void test_routing_with_az_affinity_strategy_to_1_replica() {
+        assumeTrue(SERVER_VERSION.isGreaterThanOrEqualTo("8.0.0"), "Skip for versions below 8");
+
+        String az = "us-east-1a";
+        int nGetCalls = 3;
+        String getCmdstat = String.format("cmdstat_get:calls=%d", nGetCalls);
+
+        GlideClusterClient configSetClient =
+                GlideClusterClient.createClient(commonClusterClientConfig().requestTimeout(2000).build())
+                        .get();
+
+        // reset availability zone for all nodes
+        configSetClient.configSet(Map.of("availability-zone", ""), ALL_NODES).get();
+        assertEquals(configSetClient.configResetStat().get(), OK);
+
+        Long fooSlotKey =
+                (Long)
+                        configSetClient
+                                .customCommand(new String[] {"CLUSTER", "KEYSLOT", "foo"})
+                                .get()
+                                .getSingleValue();
+        int convertedKey = Integer.parseInt(fooSlotKey.toString());
+        configSetClient
+                .configSet(
+                        Map.of("availability-zone", az),
+                        new RequestRoutingConfiguration.SlotIdRoute(convertedKey, REPLICA))
+                .get();
+        configSetClient.close();
+
+        GlideClusterClient azTestClient = createAzTestClient(az);
+
+        // execute GET commands
+        for (int i = 0; i < nGetCalls; i++) {
+            azTestClient.get("foo").get();
+        }
+
+        ClusterValue<String> infoResult =
+                azTestClient.info(new InfoOptions.Section[] {InfoOptions.Section.ALL}, ALL_NODES).get();
+        Map<String, String> infoData = infoResult.getMultiValue();
+
+        // Check that all replicas have the same number of GET calls
+        long matchingEntries =
+                infoData.values().stream()
+                        .filter(value -> value.contains(getCmdstat) && value.contains(az))
+                        .count();
+        assertEquals(1, matchingEntries);
+        azTestClient.close();
+    }
+
+    /**
+     * Test that the client with az affinity strategy will only route to the 1 replica with the same
+     * az.
```

Comment:
Is this supposed to be the exact same comment as the test above, or is this a copy & paste error?